src/main/java/org/checkstyle/autofix/recipe/HexLiteralCase.java

```diff
@@ -0,0 +1,149 @@
+        private static final int A_ASCII = 97;
+
+        /** ASCII value for lowercase 'f'. */
+        private static final int F_ASCII = 102;
+
+        /** Difference between lowercase and uppercase ASCII letters. */
+        private static final int ASCII_CASE_DIFFERENCE = 32;
+
+        private Path sourcePath;
+
+        @Override
+        public J.CompilationUnit visitCompilationUnit(
+                J.CompilationUnit cu, ExecutionContext executionContext) {
+            this.sourcePath = cu.getSourcePath().toAbsolutePath();
+            return super.visitCompilationUnit(cu, executionContext);
+        }
+
+        @Override
+        public J.Literal visitLiteral(J.Literal literal, ExecutionContext executionContext) {
+            J.Literal result = super.visitLiteral(literal, executionContext);
+            final String valueSource = result.getValueSource();
+
+            if (shouldProcessLiteral(result, valueSource)) {
+                final String newValueSource = convertLowercaseHexToUppercase(valueSource);
+                if (newValueSource != null) {
+                    result = result.withValueSource(newValueSource);
+                }
+            }
+
+            return result;
+        }
+
+        /**
+         * Determines whether the given literal should be processed based on its type and format.
+         *
+         * @param literal the {@link J.Literal} node to check
+         * @param valueSource the source value of the literal as a string
+         * @return {@code true} if the literal meets the criteria for processing,
+         *     {@code false} otherwise
+         */
+        private boolean shouldProcessLiteral(J.Literal literal, String valueSource) {
+            return valueSource != null
+                    && (valueSource.startsWith("0x") || valueSource.startsWith("0X"))
+                    && (literal.getType() == JavaType.Primitive.Long
+                    || literal.getType() == JavaType.Primitive.Int)
+                    && isAtViolationLocation(literal);
+        }
+
+        /**
+         * Converts any lowercase hexadecimal letters (a-f) to uppercase (A-F).
+         *
+         * @param valueSource the original literal source
+         * @return a new uppercase version if modified, or {@code null} if no changes were made
+         */
+        private String convertLowercaseHexToUppercase(String valueSource) {
+            final char[] chars = valueSource.toCharArray();
+            boolean changed = false;
+
+            for (int currChar = 0; currChar < chars.length; currChar++) {
+                final char c = chars[currChar];
+                if (c >= A_ASCII && c <= F_ASCII) {
+                    chars[currChar] = (char) (c - ASCII_CASE_DIFFERENCE);
+                    changed = true;
+                }
+            }
+            String result = null;
+            if (changed) {
+                result = new String(chars);
+            }
+            return result;
+        }
```

Comment:
considering we arleady know that `valueSource` starts with `0x or 0X` at this point, can we simplify the logic in this function to something like:
```
String prefix = valueSource.substring(0, 2); // we can make "0x" as a constant and replace 2 with its length, also reuse this constant in `shouldProcessLiteral`
String result = prefix + valueSource.substring(prefix.length()).toUpperCase(Locale.ROOT);
```

Reply:
> @rdiachenko , should we mention this in our readme, as it drastic difference in design in comparison to all other openrewire recepies in web ?

yes, https://github.com/checkstyle/checkstyle-openrewrite-recipes/issues/120